src/client/datascience/interactive-common/interactiveBase.ts

```
@@ -600,8 +600,12 @@ export abstract class InteractiveBase extends WebViewHost<IInteractiveWindowMapp
                 await this.setLaunchingFile(file);
 
                 if (debug) {
-                    // Attach our debugger
-                    await this.jupyterDebugger.startDebugging(this._notebook);
+                    // Attach our debugger based on run by line setting
+                    if (debug.runByLine && debug.hashFileName) {
+                        await this.jupyterDebugger.startRunByLine(this._notebook, debug.hashFileName);
+                    } else {
+                        await this.jupyterDebugger.startDebugging(this._notebook);
```

runByLine = true without a debug.hasFileName just goes down the normal debugging path, which seems a bit funny. Is that intended?

---

No it should probably throw an exception.

---
In reply to: [420478342](https://github.com/microsoft/vscode-python/pull/11608#discussion_r420478342) [](ancestors = 420478342)